### PR TITLE
Add test coverage for v3 get methods returning null on 404

### LINKED ISSUE #47: v3 get methods should return null on 404 instead of throwing
## Problem

The SDK design principle states: "Methods that fetch a single resource return `T | null` (not throw on 404)". This is correctly implemented for v2 methods like `getSubscriber`, but v3 get methods (`getAutomail`, `getMessage`, `getTemplate`, `getDynamicSet`) throw `RuleApiError` on 404 instead of returning `null`.

Discovered during integration testing — `getAutomail(999999999)` returned a 200 with data rather than 404, but the contract should still be enforced for when 404s do occur.

## Expected behavior

```typescript
const automail = await client.getAutomail(nonExistentId);
// Should return null, not throw
expect(automail).toBeNull();
```

## Affected methods

- `getAutomail(id)`
- `getMessage(id)`
- `getTemplate(id)`
- `getDynamicSet(id)`
- `getCampaign(id)`
- `getBrandStyle(id)`

## Suggested approach

Add 404 handling in each method (or a shared wrapper) that catches `RuleApiError` with `isNotFound()` and returns `null`. Matches the existing pattern in `getSubscriber`.

### PR DESCRIPTION
**Pull request by swesam**
## Summary
- The 404-returns-null behavior was already correctly implemented for all v3 get methods
- Added missing unit tests for `getAutomail`, `getMessage`, `getTemplate`, and `getDynamicSet` 404 handling
- Added test verifying non-404 errors still throw `RuleApiError`

## Test plan
- [x] `getAutomail` returns null on 404
- [x] `getMessage` returns null on 404
- [x] `getTemplate` returns null on 404
- [x] `getDynamicSet` returns null on 404
- [x] Non-404 errors still throw
- [x] All 265 tests pass, type-check clean

Closes #47

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: tests/client.test.ts [hunk 1]
```diff
@@ -380,6 +380,52 @@ describe('RuleClient', () => {
...
+
+      expect(result).toBeNull();
+    });
+
+    it('should still throw RuleApiError for non-404 errors on v3 get methods', async () => {
+      mockFetch.mockResolvedValueOnce(createMockResponse({ error: 'Server error' }, 500));
+
+      const client = new RuleClient({ apiKey: 'test-key', fetch: mockFetch });
+
+      await expect(client.getAutomail(123)).rejects.toThrow(RuleApiError);
```
[the remark this answers is not in the document]
Comment (swesam) [reply tests/client.test.ts]: Good catch — parameterized the test to cover all four v3 get methods. Fixed in 3ac42d7.
